### DISCUSSION TITLE
A router cannot stop when it has circular references.

### DISCUSSION THREAD

**Discussion by ofk** [fastapi/fastapi#11424] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import APIRouter, FastAPI

app = FastAPI()
api = APIRouter(prefix="/api")

@api.post("/foobar")
def foobar():
    return {"foo": "bar"}

# I made a typo where I should have written `app.include_router(api)`.
api.include_router(api)
```

### Description

Because of the circular reference caused by `api.include_router(api)`, it is desirable for FastAPI to promptly raise an error and terminate.

However, when executing this code, it becomes impossible to exit the infinite loop that uvicorn falls into.

### Operating System

macOS

### FastAPI Version

0.110.1

### Pydantic Version

2.6.4

### Python Version

Python 3.12.2

### Additional Context

If your server doesn't start in an infinite loop, there's a possibility that the phenomenon described in this discussion is occurring.

**Comment by JavierSanchezCastro** [comment, marked as answer]
I've created a PR #11425, is that what are you looking for? (It's not working, but to get the idea)

**Reply by ofk** [reply]
Thank you!
That's the assertion functionality I'm looking for. What can I do to help get this merged?

**Reply by JavierSanchezCastro** [reply]
We need to create some tests, if you want to add them yourself that's fine, otherwise, I'll try to do it over these weeks
